docs/src/main/asciidoc/spring-data-jpa-guide.adoc

```diff
@@ -0,0 +1,592 @@
+    -DclassName="org.acme.spring.data.jpa.FruitResource" \
+    -Dpath="/greeting" \
+    -Dextensions="spring-data-jpa,jaxrs-json,quarkus-jdbc-postgresql"
+----
+
+This command generates a Maven project with a REST endpoint and imports the `spring-data-jpa` extension.
+
+== Define the Entity
+
+Throughout the course of this guide, we will be using the following JPA Entity:
+
+[source,java]
+----
+package org.acme.spring.data.jpa;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+@Entity
+public class Fruit {
+
+    @Id
+    @GeneratedValue
+    private Long id;
+
+    private String name;
+
+    private String color;
+
+
+    public Fruit() {
+    }
+
+    public Fruit(String name, String color) {
+        this.name = name;
+        this.color = color;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+}
+----
+
+
+== Configure database access properties
+
+Add the following properties to `application.properties` to configure access to a local PostgreSQL.
+
+[source,properties]
+----
+quarkus.datasource.url=jdbc:postgresql:quarkus_test
+quarkus.datasource.driver=org.postgresql.Driver
+quarkus.datasource.username=quarkus_test
+quarkus.datasource.password=quarkus_test
+quarkus.datasource.max-size=8
+quarkus.datasource.min-size=2
+quarkus.hibernate-orm.database.generation=drop-and-create
+----
+
+This configuration assumes that PostgreSQL will be running locally.
+
+A very easy way to accomplish that is by using the following docker command:
+
+```
+docker run --ulimit memlock=-1:-1 -it --rm=true --memory-swappiness=0 --name quarkus_test -e POSTGRES_USER=quarkus_test -e POSTGRES_PASSWORD=quarkus_test -e POSTGRES_DB=quarkus_test -p 5432:5432 postgres:10.5
+```
+
+If you plan on using a different setup, please change  `application.properties` accordingly.
+
+== Prepare the data
+
+To make it easier to showcase some capabilities of Spring Data JPA on Quarkus, we will ensure that the database contains some test data.
+That is done by adding the following content to a new file named `src/main/resources/import.sql`:
+
+[source,sql]
+----
+INSERT INTO fruit(id, name, color) VALUES (1, 'Cherry', 'Red');
+INSERT INTO fruit(id, name, color) VALUES (2, 'Apple', 'Red');
+INSERT INTO fruit(id, name, color) VALUES (3, 'Banana', 'Yellow');
+INSERT INTO fruit(id, name, color) VALUES (4, 'Avocado', 'Green');
+----
+
+Hibernate will execute these queries on application startup.
+
+== Define the repository
+
+It is now time to define the repository we will use to access `Fruit`. Create a new interface like so:
```

Comment:
I think it should be clearer that it's really standard Spring Data JPA.

Reply:
Does it sound better now?